### ISSUE TITLE
Darkmode for Web Interface

### ISSUE THREAD

**Issue by onlynow2** [gramps-project/Gramps.js#145]
It would be nice, having the option for toggeling light / dark mode.
Possible location would be in the settings menu or next to the search / profile icon top right.

**Comment by DavidMStraub** [comment]
I agree this would be something that some users might find useful. Personally, I will not work on it since it's a huge amount of work for a feature I will not use. If someone wants to tackle this, please self-assign and I can provide guidance if needed.

**Comment by zeitchef** [comment]
I might be interested in looking into this. @DavidMStraub Are there any guidelines regarding dark mode design tokens, or would this still need to be sorted?

**Comment by DavidMStraub** [comment]
Hi, that would be great! There are no guidelines yet. However, over the next month, I expect we will slowly start migrating the components from the legacy material components v0.25 to the new material web v1.0, which has buil-in [support for themes](https://material-web.dev/theming/material-theming/) as far as I can see. Perhaps we can already start using those conventions? Does that help?

At the moment, the main colors are set here using the CSS variables of the old material components:
https://github.com/gramps-project/Gramps.js/blob/main/src/SharedStyles.js#L10-L15

However, there are still many components that have hard-coded colors (e.g. shades of gray), that will have to be adapted.

Some suggestions for implementing the dark mode switch:

- We can store the user preference in local storage using the existing functions `updateSettings` and `getSettings`
- I suggest to put the theme selection on the user settings page where also the language selector is located

**Comment by zeitchef** [comment]
Hi @DavidMStraub - I've got some time this week and would like to dig into this a bit. I've hit my first glitch though, after setting up a dev environment, I can't seem to register a test user - every combination I've tried just returns a `409` error. Is there a dedicated test user I can use to login with, or a pattern I can use to register one? Thanks.

**Comment by DavidMStraub** [comment]
(Confusingly, in the new Material Web, components and CSS variables start with `md-`, while in the old version components started with `mwc-` and CSS variables with `mdc-`.)

**Comment by tismofied** [comment]
Has there been any progress made on this?

**Comment by DavidMStraub** [comment]
No.

**Comment by joelabc** [comment]
Would be a nice to have NGL

**Comment by DavidMStraub** [comment]
Implemented in #755!
